src/hypothesis/extra/django/models.py

```diff
@@ -137,26 +136,19 @@ def _get_strategy_for_field(f):
         if isinstance(f, (dm.CharField, dm.TextField)) and f.blank:
             choices.append(u'')
         strategy = st.sampled_from(choices)
-    elif isinstance(f, dm.EmailField):
-        return emails
+    elif isinstance(f, dm.SlugField):
+        strategy = st.text(alphabet=safe_letters,
+                           min_size=(None if f.blank else 1),
+                           max_size=f.max_length)
     elif type(f) in (dm.TextField, dm.CharField):
         strategy = st.text(min_size=(None if f.blank else 1),
                            max_size=f.max_length)
-    elif type(f) == dm.DecimalField:
-        m = 10 ** f.max_digits - 1
-        div = 10 ** f.decimal_places
-        q = Decimal('1.' + ('0' * f.decimal_places))
-        strategy = (
-            st.integers(min_value=-m, max_value=m)
-            .map(lambda n: (Decimal(n) / div).quantize(q)))
+    elif isinstance(f, dm.DecimalField):
+        bound = Decimal(10 ** f.max_digits - 1) / (10 ** f.decimal_places)
+        strategy = st.decimals(min_value=-bound, max_value=bound,
+                               places=f.decimal_places)
     else:
-        try:
-            strategy = field_mappings()[type(f)]
-        except KeyError:
-            if f.null:
-                return None
-            else:
-                raise UnmappedFieldError(f)
+        strategy = field_mappings().get(type(f), st.nothing())
```

Comment:
Nope - it's checked in exactly the same place that the exception was caught, and added to the mandatory set in the same way as before.  I was quite careful to ensure that this doesn't change behaviour, just simplify the control flow and handling of nullable fields.

- A known field has the same handling.
- An unknown, nullable field now returns an explicit `none()` strategy instead of omitting the argument to the model builder (this may be worth reverting for efficiency, but it's not a loss of information).  Previously, returns None and is skipped.
- An unknown non-nullable field returns `nothing()` and is added to the mandatory set.  Previously, raised UnmappedField from KeyError; this was caught and the field name added to the mandatory set.